libraries/WiFi/examples/WiFiProv/README.md

```diff
@@ -0,0 +1,149 @@
+* prov_event_handler - provisioning event handler callback
+* user_data
+
+## API to be called before WiFi.beginProvision()
+
+* nvs_flash_init()
+* tcpip_adapter_init()
+
+## Necessary callbacks
+
+* WiFi event handler [ WiFiEvent ]
+* Provisioning event handler [ prov_event_handler ]
+
+### About structure [ prov_config_t ]
+
+* prov_scheme : choose the mode of transfer
+    * softap
+    * BLE
+* security : choose whether security to apply or not
+    * WIFI_PROV_SECURITY_1
+    * WIFI_PROV_SECURITY_0
+* pop : if security is choosen [ WIFI_PROV_SECURITY_1 ] then specify proof of possesion
+* scheme_event_handler : specify the handlers according to the mode choosen
+    * BLE :
+        - WIFI_PROV_SCHEME_BLE_EVENT_HANDLER_FREE_BTDM
+        - WIFI_PROV_SCHEME_BLE_EVENT_HANDLER_FREE_BLE
+        - WIFI_PROV_SCHEME_BLE_EVENT_HANDLER_FREE_BT
+    * SoftAp :
+        - WIFI_PROV_EVENT_HANDLER_NONE
+
+* service_name : Specify service name for ESP_Station while provisioning via if it is not specified then default choosen name if WIFI_XXXX and for BLE service name is BLE_XXXX where XXXX is the last 4 digits of the boards MAC. 
+
+* service_key : Specify service key while provisioning if choosen mode of provisioning is BLE then service_key is always NULL
+
+* uuid : user can specify there own 16 bit uuid while provisioning using BLE if not specified then default value is taken
+        - {  0xb4, 0xdf, 0x5a, 0x1c, 0x3f, 0x6b, 0xf4, 0xbf,
+             0xea, 0x4a, 0x82, 0x03, 0x04, 0x90, 0x1a, 0x02, }
+
+## Hardware Requirement
+
+* Enable Debuger : [ Tools -> Core Debug Level ]
+
+## Example output
+
+### Credentials are available on device
+
+```
+load:0x3fff0018,len:4
+load:0x3fff001c,len:1216
+ho 0 tail 12 room 4
+load:0x40078000,len:10864
+load:0x40080400,len:6432
+entry 0x400806b8
+
+[I][WiFiProv.cpp:56] wifi_init_sta(): 
+Aleardy Provisioned, starting Wi-Fi STA
+[I][WiFiProv.cpp:57] wifi_init_sta(): 
+CONNECTING ACCESS POINT CREDENTIALS : 
+[I][WiFiProv.cpp:60] wifi_init_sta(): 
+SSID : GIONEE M2
+Password : 123456789
+Connected with IP address : 192.168.43.120
+
+```
+
+### Provisioning using SOFTAP
+
+```
+Rebooting...
+
+ets Jun  8 2016 00:22:57
+rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
+configsip: 0, SPIWP:0xee
+clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
+mode:DIO, clock div:1
+load:0x3fff0018,len:4
+load:0x3fff001c,len:1216
+ho 0 tail 12 room 4
+load:0x40078000,len:10864
+load:0x40080400,len:6432
+entry 0x400806b8
+
+[I][WiFiProv.cpp:133] beginProvision(): 
+Starting station using SOFTAP
+ service_name : WIFI_XXXX
+ password : 123456789
+ pop : abcd1234
+
+Provisioning started
+Give Credentials of your access point using " Android app "
+
+Provisioning Successful
+
+Connected with IP address : 192.168.43.120
+Received Wi-Fi credentials
+	SSID : GIONEE M2
+	Password : 123456789
+Provisioning Ends
+
+```
+
+### Provisioning using BLE
+
+```
+d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
```

Comment:
no need to have the bootloader code. that can change and is not really the point of the topic.